rest-server/src/controllers/user.js

```diff
@@ -15,67 +15,28 @@
 // DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-
 // module dependencies
-const jwt = require('jsonwebtoken');
-const authConfig = require('../config/auth');
-const authModel = require('../models/auth');
+const userModel = require('../models/user');
 const logger = require('../config/logger');
 
-
-/**
- * Login.
- */
-const login = (req, res) => {
-  const username = req.body.username;
-  const password = req.body.password;
-  const expiration = req.body.expiration;
-  authModel.check(username, password, (err, state, admin) => {
-    if (err || !state) {
-      logger.warn('user %s authentication failed', username);
-      return res.status(401).json({
-        error: 'AuthenticationFailed',
-        message: 'authentication failed'
-      });
-    } else {
-      jwt.sign({
-        username: username,
-        admin: admin
-      }, authConfig.secret, { expiresIn: expiration }, (signError, token) => {
-        if (signError) {
-          logger.warn('sign token error\n%s', signError.stack);
-          return res.status(500).json({
-            error: 'SignTokenFailed',
-            message: 'sign token failed'
-          });
-        }
-        return res.json({
-          token,
-          user: username
-        });
-      });
-    }
-  });
-};
-
 /**
- * Update user.
+ * Create / update a user.
  */
 const update = (req, res) => {
   const username = req.body.username;
   const password = req.body.password;
   const admin = req.body.admin;
   const modify = req.body.modify;
   if (req.user.admin) {
-    authModel.update(username, password, admin, modify, (err, state) => {
+    userModel.update(username, password, admin, modify, (err, state) => {
       if (err || !state) {
         logger.warn('update user %s failed', username);
         return res.status(500).json({
           error: 'UpdateFailed',
           message: 'update failed'
         });
       } else {
-        return res.json({
+        return res.status(201).json({
```

Comment:
why not 202?

Reply:
According to http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html, 201 means you get the resource immediately while 202 means the request is successfully queued. So I think it should be 201 here.